### PR TITLE
Feature/adding more file formats for duckdb

### PR DESCRIPTION
**Pull request by PaoloLeonard**
Adding support for csv, parquet and json file in Soda-core for the duckdb engine.

### CODE REVIEW

File: soda/duckdb/tests/test_duckdb.py [hunk 1]
```diff
@@ -21,3 +23,98 @@ def test_pandas_df(data_source_fixture: DataSourceFixture):
...
+    scan.set_data_source_name("csv_dataset")
+    scan.add_configuration_yaml_str(
+        f"""
+          data_source csv_dataset:
+            type: duckdb
+            path: {csv_path}
+        """
+    )
+    scan.add_sodacl_yaml_str(
+        f"""
```
Comment (m1n0) [inline soda/duckdb/tests/test_duckdb.py:48]: Can we remove the f-strings as there are no vars in the string and code analysis is complaining?

```suggestion
        """
```

File: soda/duckdb/tests/test_duckdb.py [hunk 2]
```diff
@@ -21,3 +23,98 @@ def test_pandas_df(data_source_fixture: DataSourceFixture):
...
+    scan.set_data_source_name("json_dataset")
+    scan.add_configuration_yaml_str(
+        f"""
+          data_source json_dataset:
+            type: duckdb
+            path: {json_path}
+        """
+    )
+    scan.add_sodacl_yaml_str(
+        f"""
```
Comment (m1n0) [inline soda/duckdb/tests/test_duckdb.py:80]:
```suggestion
        """
```

File: soda/duckdb/tests/test_duckdb.py [hunk 3]
```diff
@@ -21,3 +23,98 @@ def test_pandas_df(data_source_fixture: DataSourceFixture):
...
+    scan.set_data_source_name("parquet_dataset")
+    scan.add_configuration_yaml_str(
+        f"""
+          data_source parquet_dataset:
+            type: duckdb
+            path: {parquet_path}
+        """
+    )
+    scan.add_sodacl_yaml_str(
+        f"""
```
Comment (m1n0) [inline soda/duckdb/tests/test_duckdb.py:111]:
```suggestion
        """
```

**Review by m1n0 — CHANGES_REQUESTED** [review]
Thanks for the contribution! I added a small suggestion otherwise this LGTM!

**Comment by PaoloLeonard** [comment]
> Thanks for the contribution! I added a small suggestion otherwise this LGTM!

Thanks for the suggestions, applied!

**Review by m1n0 — APPROVED** [review]

**Pull request merged**